flow/activities/flowable.go

```diff
@@ -774,58 +785,116 @@ func (a *FlowableActivity) RecordSlotSizes(ctx context.Context) error {
-			connCtx := context.WithValue(ctx, internal.FlowMetadataKey, flowMetadata)
-			logger = internal.LoggerFromCtx(connCtx)
-			srcConn, err := connectors.GetByNameAs[connectors.CDCPullConnector](connCtx, nil, a.CatalogPool, config.SourceName)
+			ctx = context.WithValue(ctx, internal.FlowMetadataKey, flowMetadata)
+			logger = internal.LoggerFromCtx(ctx)
+			srcConn, err := connectors.GetByNameAs[connectors.CDCPullConnector](ctx, nil, a.CatalogPool, info.config.SourceName)
 			if err != nil {
 				if !errors.Is(err, errors.ErrUnsupported) {
 					logger.Error("Failed to create connector to handle slot info", slog.Any("error", err))
 				}
 				return
 			}
-			defer connectors.CloseConnector(connCtx, srcConn)
+			defer connectors.CloseConnector(ctx, srcConn)
 
-			slotName := "peerflow_slot_" + config.FlowJobName
-			if config.ReplicationSlotName != "" {
-				slotName = config.ReplicationSlotName
+			slotName := "peerflow_slot_" + info.config.FlowJobName
+			if info.config.ReplicationSlotName != "" {
+				slotName = info.config.ReplicationSlotName
 			}
-			peerName := config.SourceName
+			peerName := info.config.SourceName
 
-			activity.RecordHeartbeat(connCtx, fmt.Sprintf("checking %s on %s", slotName, peerName))
-			if connCtx.Err() != nil {
+			activity.RecordHeartbeat(ctx, fmt.Sprintf("checking %s on %s", slotName, peerName))
+			if ctx.Err() != nil {
 				return
 			}
+			status, sErr := internal.GetWorkflowStatus(ctx, a.CatalogPool, nil, info.workflowID)
+			if sErr != nil {
+				logger.Error("Failed to get workflow status", slog.Any("error", err))
+			}
+			info.status = status
+
 			if a.OtelManager != nil {
-				a.OtelManager.Metrics.SyncedTablesGauge.Record(connCtx, int64(len(config.TableMappings)), metric.WithAttributeSet(
-					attribute.NewSet(
-						attribute.String(otel_metrics.FlowNameKey, config.FlowJobName),
-						attribute.String(otel_metrics.PeerNameKey, peerName),
-						attribute.String(otel_metrics.SourcePeerType, fmt.Sprintf("%T", srcConn)),
-					)))
+				a.OtelManager.Metrics.SyncedTablesGauge.Record(ctx, int64(len(info.config.TableMappings)))
+				a.OtelManager.Metrics.FlowStatusGauge.Record(ctx, 1, metric.WithAttributeSet(attribute.NewSet(
+					attribute.String(otel_metrics.FlowStatusKey, status.String()),
+				)))
 			}
-			if err := srcConn.HandleSlotInfo(connCtx, a.Alerter, a.CatalogPool, &alerting.AlertKeys{
-				FlowName: config.FlowJobName,
+			if err := srcConn.HandleSlotInfo(ctx, a.Alerter, a.CatalogPool, &alerting.AlertKeys{
+				FlowName: info.config.FlowJobName,
 				PeerName: peerName,
 				SlotName: slotName,
 			}, slotMetricGauges); err != nil {
 				logger.Error("Failed to handle slot info", slog.Any("error", err))
 			}
-		}()
+		}(ctx)
+	}
+	if a.OtelManager != nil {
+		operationalFlows := filterOperationalFlowStatuses(infos)
+		operationFlowCount := len(operationalFlows)
+		a.OtelManager.Metrics.ActiveFlowsGauge.Record(ctx, int64(operationFlowCount))
+		for _, info := range operationalFlows {
```

Comment:
why are we looping over operationalFlows here? we aren't using `info` outside of `a.GetFlowMetadata`

Reply:
the metric needs to be only emitted for operation/active flows.
basically `cpu_limit/active_flows` for every active flow

Reply:
You can lift computing the metric values out of the loop

Reply:
make sense

Reply:
I'm also debating whether I should pull the data from cgroups or just use simple env vars which are injected at runtime via helm chart

Reply:
I went ahead with injected env